frontend/appflowy_tauri/src/appflowy_app/components/layout/NavigationPanel/NewFolderButton.hooks.ts

```diff
@@ -1,12 +1,28 @@
-import { useAppDispatch } from '../../../stores/store';
+import { useAppDispatch, useAppSelector } from '../../../stores/store';
 import { foldersActions } from '../../../stores/reducers/folders/slice';
 import { nanoid } from 'nanoid';
+import { CreateAppPayloadPB, FolderEventCreateApp } from '../../../../services/backend/events/flowy-folder';
+import { workspaceSlice } from '../../../stores/reducers/workspace/slice';
 
 export const useNewFolder = () => {
   const appDispatch = useAppDispatch();
+  const workspace = useAppSelector((state) => state.workspace);
 
-  const onNewFolder = () => {
-    appDispatch(foldersActions.addFolder({ id: nanoid(8), title: 'New Folder 1' }));
+  const onNewFolder = async () => {
```

Comment:
I think OnNewFolder should be moved to the addFolder handler. Then the UI will wait for the response comes from the backend and handler the errors if there are any.

![image](https://user-images.githubusercontent.com/86001920/220862885-910d60da-51b3-47df-acf1-5a1a258e5664.png).

Reply:
refactored to use backend service.
To complete error handling we need to make a window that can show error messages. That will be a different task that I will create soon.
Right here addFolder is in correct place because it is called only on success.